### DISCUSSION TITLE
Formatters associated with hard-coded language vs configurable file-type

### DISCUSSION THREAD

**Discussion by TaquiRaza** [microsoft/vscode-discussions#492] [Extension Development QnA] [answered]
To demonstrate the issue, install the `snowflake` extension (for using the snowflake cloud database) which will associate `.sql` files with the `snowflake-sql` language. This extension does not support code formatting.

Next install a sql-code formatter which is associated with the `sql` language. It is impossible to use it to format a `snowflake-sql` file. The only way is to change the associated language to sql, format it, then change the association back to snowflake-sql.

It would be nice to associate the formatter with the file type instead of the language (but I understand why that wasn't the design choice), or else override the extensions associated language in settings.  I've tried editing the package.json of the local extension to be activated `onLanguage: snowflake-sql` but to no effect.

**Comment by r3m0t** [comment, marked as answer]
The snowflake developers chose to use a new language to avoid getting behaviours from the SQL language. If you support both languages you can register your formatter as supproting both. `registerDocumentFormattingEditProvider([ {language: "sql"}, {language: "snowflake-sql"}], ...)`
